src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -100,12 +100,12 @@ def __init__(
         # When `is_temp_table_for_cleanup` is True, it's a temp table
         # generated by Snowpark (currently only df.cache_result) under the hood
         # and users are not aware of it.
-        if is_temp_table_for_cleanup:
-            session._temp_table_ref_count_map[name] += 1
+        if is_temp_table_for_cleanup and session._temp_table_cleaner is not None:
+            session._temp_table_cleaner.inc_ref_count(name)
             # the finalizer will be triggered when it gets garbage collected
             # and this table will be dropped finally
             self._finalizer = weakref.finalize(
-                self, session._dec_temp_table_ref_count, name
+                self, session._temp_table_cleaner.dec_ref_count, name
```

Comment:
if we call the other interface table_ref_created, maybe let's call this table_ref_destroyed to be consistent, this two interfaces basically defines the actions needs to be take when those two events happens.

Reply:
table_ref_destroyed is not correct because after decrement if the reference count is not 0, the temp table won't be dropped. I'd prefer to keep the current name because it describes what it's exactly doing, and I also make it private now and won't be used outside of class, which might be confusing.

Reply:
removed inc_ref_count, and change the name of this method to `_delete_ref_count`